src/Files.App/Views/NavigationInteractionTracker.cs

```diff
@@ -0,0 +1,205 @@
+		{
+			_rootElement = rootElement;
+			_backIcon = backIcon;
+			_forwardIcon = forwardIcon;
+
+			ElementCompositionPreview.SetIsTranslationEnabled(_backIcon, true);
+			ElementCompositionPreview.SetIsTranslationEnabled(_forwardIcon, true);
+			_rootVisual = ElementCompositionPreview.GetElementVisual(_rootElement);
+			_backVisual = ElementCompositionPreview.GetElementVisual(_backIcon);
+			_forwardVisual = ElementCompositionPreview.GetElementVisual(_forwardIcon);
+
+			_props = _rootVisual.Compositor.CreatePropertySet();
+			_props.InsertBoolean("CanGoForward", false);
+			_props.InsertBoolean("CanGoBack", false);
+
+			rootElement.AddHandler(UIElement.PointerPressedEvent, new PointerEventHandler(PointerPressed), true);
+
+			SetupInteractionTracker();
+			SetupAnimations();
+		}
+
+		private void SetupInteractionTracker()
+		{
+			var compositor = _rootVisual.Compositor;
+
+			_trackerOwner = new(this);
+			_tracker = InteractionTracker.CreateWithOwner(compositor, _trackerOwner);
+			_tracker.MinPosition = new Vector3(-96f);
+			_tracker.MaxPosition = new Vector3(96f);
+
+			_source = VisualInteractionSource.Create(_rootVisual);
+			_source.ManipulationRedirectionMode = VisualInteractionSourceRedirectionMode.CapableTouchpadOnly;
+			_source.PositionXSourceMode = InteractionSourceMode.EnabledWithoutInertia;
+			_source.PositionXChainingMode = InteractionChainingMode.Auto;
+			_source.PositionYSourceMode = InteractionSourceMode.Disabled;
+			_tracker.InteractionSources.Add(_source);
+		}
+
+		private void SetupAnimations()
+		{
+			var compositor = _rootVisual.Compositor;
+
+			var backResistance = CompositionConditionalValue.Create(compositor);
+			var backResistanceCondition = compositor.CreateExpressionAnimation("tracker.Position.X < 0 && tracker.Position.X > -96");
+			backResistanceCondition.SetReferenceParameter("tracker", _tracker);
+			var backResistanceValue = compositor.CreateExpressionAnimation("source.DeltaPosition.X * (1 - sqrt(1 - square((tracker.Position.X / -96) - 1)))");
+			backResistanceValue.SetReferenceParameter("source", _source);
+			backResistanceValue.SetReferenceParameter("tracker", _tracker);
+			backResistance.Condition = backResistanceCondition;
+			backResistance.Value = backResistanceValue;
+
+			var forwardResistance = CompositionConditionalValue.Create(compositor);
+			var forwardResistanceCondition = compositor.CreateExpressionAnimation("tracker.Position.X > 0 && tracker.Position.X < 96");
+			forwardResistanceCondition.SetReferenceParameter("tracker", _tracker);
+			var forwardResistanceValue = compositor.CreateExpressionAnimation("source.DeltaPosition.X * (1 - sqrt(1 - square((tracker.Position.X / 96) - 1)))");
+			forwardResistanceValue.SetReferenceParameter("source", _source);
+			forwardResistanceValue.SetReferenceParameter("tracker", _tracker);
+			forwardResistance.Condition = forwardResistanceCondition;
+			forwardResistance.Value = forwardResistanceValue;
+
```

Comment:
This is almost the same code repeated twice. You may create a method where you pass in `resistanceConditionString` and `resistanceValueString`